geode-core/src/distributedTest/java/org/apache/geode/internal/cache/RemoteTransactionDUnitTest.java

```diff
@@ -1872,12 +1858,8 @@ public Object call() {
         getCache().getTxManager().begin();
         Region r = getCache().getRegion(CUSTOMER);
         r.put(new CustId(8), new Customer("name8", "address8"));
-        try {
-          getCache().getTxManager().commit();
-          fail("expected exception that was not thrown");
-        } catch (Exception e) {
-          assertThat("AssertionError").isEqualTo(e.getCause().getMessage());
-        }
```

Comment:
This block is very confusing. There's a `fail()` which gets triggered if we don't throw an exception on calling `commit()`, but then we catch an exception and assert that it's an AssertionError (which is what gets thrown by `fail()`) and then just carry on as long as it is. This means that the only time this block doesn't result in a test failure is when we *don't* hit an exception in `commit()` so that we hit the `fail()` call and then catch and correctly assert on the exception it throws. It's worth checking to make sure, but I think this should just be:
```
getCache().getTxManager().commit();
```
with no asserts or try/catch at all.

Reply:
I think I understand what's happening here now. The "AssertionError" message is actually from the `TransactionWriterException` thrown in the `beforeCommit()` method just above this, and it's that exception that we're trying to assert on. To make things clearer, maybe the message for the exception could be extracted to a variable and changed to something generic like "exception message" and we could assert on that, rather than using hard-coded strings in both places.

We could also improve the assertions a bit by having them be more explicit:
```
assertThatThrownBy(() -> getCache().getTxManager().commit())
    .isInstanceOf(CommitConflictException.class)
    .getCause()
    .isInstanceOf(TransactionWriterException.class)
    .hasMessage(exceptionMessage);
```